lib/rumoji.rb

```diff
@@ -8,17 +8,25 @@ module Rumoji
 
   # Transform emoji into its cheat-sheet code
   def encode(str)
-    str.gsub(Emoji::ALL_REGEXP) { |match| (emoji = Emoji.find_by_string(match)) && emoji.code || match }
+    str.gsub(Emoji::ALL_REGEXP) do |match|
+      if emoji = Emoji.find_by_string(match)
+        if block_given?
+          yield emoji
+        else
+          emoji.code || match
+        end
+      end
+    end
```

Comment:
Ah, true. Nice catch. Tried to write a failing test for it but I'm not sure how to construct a string with an emoji that would match `ALL_REGEXP` but not `STRING_LOOKUP`.